src/Workspaces/Core/MSBuild/MSBuild/ProjectFile/ProjectFileInfo.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.MSBuild.Logging;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.MSBuild
+{
+    /// <summary>
+    /// Represents a project file loaded from disk.
+    /// </summary>
+    internal sealed class ProjectFileInfo
+    {
+        public bool IsEmpty { get; }
+
+        /// <summary>
+        /// The language of this project.
+        /// </summary>
+        public string Language { get; }
+
+        /// <summary>
+        /// The path to the project file for this project.
+        /// </summary>
+        public string FilePath { get; }
+
+        /// <summary>
+        /// The path to the output file this project generates.
+        /// </summary>
+        public string OutputFilePath { get; }
+
+        /// <summary>
+        /// The path to the reference assembly output file this project generates.
+        /// </summary>
+        public string OutputRefFilePath { get; }
+
+        /// <summary>
+        /// The target framework of this project.
+        /// </summary>
+        public string TargetFramework { get; }
+
+        /// <summary>
+        /// The command line args used to compile the project.
+        /// </summary>
+        public IReadOnlyList<string> CommandLineArgs { get; }
```

Comment:
>IReadOnlyList<string> [](start = 15, length = 21)

Why did you choose the `IReadOnly` types here vs. the concrete `Immutable` types? It's a stronger guarantee and lines up with the implementation of the interfaces.

Reply:
I just matched other properties in this file, which pre-existed my change. Originally, I think there were thoughts of making this a public API before even ImmutableArray<T> existed. I'll change it.